independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InterceptorInfo.java

```diff
@@ -103,6 +91,22 @@
         }
     }
 
+    private MethodInfo validateSignature(MethodInfo method) {
+        List<Type> parameters = method.parameters();
+        if (parameters.size() != 1 || !parameters.get(0).name().equals(DotNames.INVOCATION_CONTEXT)) {
```

Comment:
Shouldn't we also allow `ArcInvocationContext`?

Reply:
That's a good point but I don't think it would work right now... let me check.

Reply:
Hm, this would actually require some more work, because we also generate the interceptor beans that implement `javax.enterprise.inject.spi.Interceptor.intercept(InterceptionType, T, InvocationContext)` etc. We can create a feature request though. Currently, you would need to cast the `InvocationContext` inside the interceptor method. @manovotn WDYT?

Reply:
Fair enough, it just occurred to me when I saw the code :)

Reply:
Created https://github.com/quarkusio/quarkus/issues/15660